### PR TITLE
Add CI check to install with homebrew on `macos`

### PR DESCRIPTION
**Pull request by jtpio**
Troubleshoot https://github.com/jupyter/notebook/issues/6974

### CODE REVIEW

**Comment by jtpio** [comment]
cc @defjaf

Here are the logs for the homebrew installation, which passes on CI: https://github.com/jupyter/notebook/actions/runs/5809598910/job/15748922126?pr=7005

If you would like to double check to see if you have something similar locally.

**Comment by jtpio** [comment]
And the rest of the logs? https://github.com/jupyter/notebook/actions/runs/5809598910/job/15748922126?pr=7005

**Comment by defjaf** [comment]
I admit it's pretty hard to parse what I've got...

So I attach a file with the console output from pip. Note that this is starting from having jupyterlab 3.6.3 and notebook 6.5.5 installed, so they are upgraded to 4.0.4 and 7.0.3 (but it didn't automatically upgrade other stuff). It does seem to have put a bunch of files below `/opt/homebrew/share/jupyter/` but nothing new in `/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/share/jupyter/`. And I still get the schema console errors and the UI problems.

[pip-install-notebook==7 output.txt](https://github.com/jupyter/notebook/files/12303627/pip-install-notebook.7.output.txt)

**Comment by defjaf** [comment]
@jtpio Looking more carefully, even from the beginning of the logs, the CI is using `/Library/Frameworks/Python.framework/...` whereas my homebrew install is using `opt/homebrew/...` — it looks like the CI build isn't using the homebrew python?

**Comment by defjaf** [comment]
> Maybe we could check whether it's possible to install under /opt/homebrew on CI.

Well, I think the point is that the CI process isn't using the homebrew python and so not automatically picking up the homebrew paths like `sys.prefix`. Can you write the script with a hard-coded python executable `/opt/homebrew/bin/python3`?

**Comment by defjaf** [comment]
> Or can you write the script with a hard-coded python executable `/opt/homebrew/bin/python3`?

Of `/usr/local/bin/python3` if the CI machines are intel.

**Comment by defjaf** [comment]
Seems like the CI machines are running on Intel, and that the same issues do not arise in that case, according to https://github.com/jupyter/notebook/pull/7006.

**Comment by defjaf** [comment]
see https://github.com/jupyter/notebook/pull/7006#issuecomment-1682181317 for information about running in a venv.

**Comment by DTiming24** [comment]
Nice CI addition for the Homebrew install path. Even though it's a workflow-only change, this should make macOS breakages much easier to catch before they land on users.

**Comment by jtpio** [comment]
Closing as per https://github.com/jupyter/notebook/issues/6974#issuecomment-5324717307
